### ISSUE TITLE
MzMine 2.15: problems with "mass detection" when selects Polarity +

### ISSUE THREAD

**Issue by VicentYusa** [mzmine/mzmine2#86]
I'm working (MzMine 2.15) with raw data from Thermo (Orbitrap), adquired in ESI +, ESI -, and with fragmentation in the same run.

When I try to do "mass detection" selecting Polarity +, the program do not generate a "mass list", so when try to perform the "chromatogram builder", the program shows the warning "Error of task,...does not have a mass filter list"
However it wors when select Polarity: Any

How can I solve this problem?

Many thanks for your help

**Comment by tomas-pluskal** [comment]
Do you see the correct polarity when you "unroll" the file in the project window? Each scan should have either + or - sign.

![screen shot 2015-08-19 at 8 48 36 pm](https://cloud.githubusercontent.com/assets/1796692/9365780/b5770632-46b3-11e5-8d04-1f2690236ec1.png)

**Comment by VicentYusa** [comment]
Yes, I do
[image: Imágenes integradas 1]

2015-08-19 20:49 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
Can you send a sample data file and the exact parameters you use, so we can reproduce the problem?

**Comment by VicentYusa** [comment]
​
 DENAMIC_Luba_24042015_nepos_14.raw
<https://drive.google.com/file/d/0B4oz9dG2S_SYLUxWSTdCZ3RVX1E/view?usp=drive_web>
​Mass detector: Exact Mass
Ms Level: 1
Polarity +
Spectrum type: profile
Noise Level:1E4

2015-08-19 20:59 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by VicentYusa** [comment]
Thomas, has you received correctly the file and the information?

Thanks
Vicent

2015-08-19 20:59 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
Yes, I got it. I'll get back to you later.

**Comment by tomas-pluskal** [comment]
Strangely enough, it seems to work fine for me.

Parameters:
![screen shot 2015-08-19 at 9 24 29 pm](https://cloud.githubusercontent.com/assets/1796692/9366621/c37cd6bc-46b8-11e5-9096-1aeea9ad8022.png)

Result:
![screen shot 2015-08-19 at 8 48 36 pm](https://cloud.githubusercontent.com/assets/1796692/9366628/ce37f014-46b8-11e5-9832-82373cc7cd9e.png)

**Comment by VicentYusa** [comment]
So,  any recommendation. Maybe download again the MzMine 2.15?

2015-08-19 21:26 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
In your case no mass list is visible in the tree after running the mass detector?
Did you observe any exceptions or error messages in the MZmine console window?

**Comment by VicentYusa** [comment]
After running the mass detector, a mass list is visible, and no error
messges appear. However the file in not market with the "green signal"
The mesage of error appears after running the chromatogram builder

[image: Imágenes integradas 1]

2015-08-19 21:38 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
The "green signal" appears when all MS level 1 scans have a mass list. In your case, if you run the mass detection on the + scans and then again on - scans, you will see the green mark.

However, the Chromatogram builder module at the moment does not allow you to select a particular subset of scans. So unfortunately it will not work correctly. I will add this feature for MZmine 2.16.

**Comment by VicentYusa** [comment]
Perfect. In the meanwhile I will work with the "Any" option.

Good luck with the MzMine 2.16. You are doing a great job

Thanks again and best wishes

Vicent

2015-08-19 21:57 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
I am afraid even the "Any" option is not going to work for this file, because the algorithm will try to build chromatograms from the positive and negative scans mixed together. Obviously, you will end up with a mess, or just with an empty peak list.

You may need to separate the + and - scans into separate files, e.g. using ProteoWizard.

**Comment by VicentYusa** [comment]
Woof !!!!
This are not good news. In fact I've been working with many similar files.

So the final List of features, when you export to a csv file , are not
"real m/z"?

2015-08-19 22:05 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
Well, the m/z values are real (in the sense they come from your data) but they cannot really form chromatograms, so basically any further processing is nonsense in my view.

Better for you to wait for MZmine 2.16 or use ProteoWizard to separate the + and - scans into individual files.

**Comment by VicentYusa** [comment]
Ok. Thanks so much for the advice; ..and do not delay in building Mzmine
2.16

Best whishes


Vicent



2015-08-19 22:21 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by VicentYusa** [comment]
Hi Tomas

It seems that proteoWizard can't upload this files. Do you know why?

Thanks
Vicent

2015-08-19 20:49 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
ProteoWizard does not "upload" files. But you can use the msconvert tool from ProteoWizard to separate the scans:

```
msconvert thermo.raw -o destination_directory --outfile file_name --filter "polarity positive"
```

**Comment by VicentYusa** [comment]
Hi Tomas,
The problem is that I can't find the "polarity filter" in the List of Filters of MsConvertor. Do you know how to add it to the List?
Many thanks in advance
Vicent

**Comment by tomas-pluskal** [comment]
Use the command line version of msconvert, not the GUI version.

**Comment by VicentYusa** [comment]
Hi Tomas,

After working with Msconvert, something has happened inside the Command
Promt.
So now the MzMine 2.15 is not working. I have instaled again, but the same
Error still appear. Do you know how to solve it?
Thanks
Vicent
[image: Imágenes integradas 1]

2015-08-20 10:58 GMT+02:00 Tomáš Pluskal <notifications@github.com>:

**Comment by tomas-pluskal** [comment]
I cannot see the attachment. Please use the github web interface to post images (https://github.com/mzmine/mzmine2/issues/86).

**Comment by tomas-pluskal** [comment]
I added the support for polarity selection to Chromatogram builder. It will be included in upcoming version 2.16. Closing this issue now.

**Comment by VicentYusa** [comment]
Thanks Tomas for your help.
I'm sure this new function will be well received by the MzMine users
Good luck with the version 2.16

Vicent
